osu.Server.Queues.ScoreStatisticsProcessor/Commands/Maintenance/VerifyImportedScoresCommand.cs

```diff
@@ -286,31 +238,54 @@ public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
                     }
                 }
 
-                if (elasticItems.Count > 0)
-                {
-                    if (!DryRun)
-                        elasticQueueProcessor.PushToQueue(elasticItems.ToList());
-                    Console.WriteLine($"Queued {elasticItems.Count} items for indexing");
-                }
-
-                Console.SetCursorPosition(0, Console.GetCursorPosition().Top);
                 Console.Write($"Processed up to {importedScores.Max(s => s.id)} ({deleted} deleted {fail} failed)");
+                Console.SetCursorPosition(0, Console.GetCursorPosition().Top);
 
                 lastId += (ulong)BatchSize;
+                flush(conn);
             }
 
+            flush(conn, true);
+
             return 0;
         }
 
+        private void flush(MySqlConnection conn, bool force = false)
+        {
+            int bufferLength = sqlBuffer.Length;
+
+            if (bufferLength == 0)
+                return;
+
+            if (bufferLength > 1024 || force)
+            {
+                if (!DryRun)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine($"Flushing sql batch ({bufferLength:N0} bytes)");
+                    conn.Execute(sqlBuffer.ToString());
+
+                    if (elasticItems.Count > 0)
+                    {
+                        if (!DryRun)
+                            elasticQueueProcessor.PushToQueue(elasticItems.ToList());
+                        Console.WriteLine($"Queued {elasticItems.Count} items for indexing");
+                    }
```

Comment:
Pushing to elastic here is potentially wrong. This is because `elasticItems` is not cleared here; it is cleared in the main loop. So there is a possibility of the following happening:

1. Batch 1 is verified. Items are added to `elasticItems` for elastic processing as this is happening.
2. Batch 1 is flushed. If the accumulated SQL buffer length is below 1024 bytes, none of this executes, so the items aren't pushed for processing to elastic.
3. Batch 2 is verified. At this point `elasticItems` is cleared in the main processing loop. The items in that list cannot be recovered again and will never be pushed to elastic even when the relevant buffered SQL statements from step (2) execute.